zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/DecisionDefinitionController.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller;
+
+import io.camunda.service.DecisionDefinitionServices;
+import io.camunda.service.search.query.DecisionDefinitionQuery;
+import io.camunda.zeebe.gateway.protocol.rest.DecisionDefinitionSearchQueryRequest;
+import io.camunda.zeebe.gateway.protocol.rest.DecisionDefinitionSearchQueryResponse;
+import io.camunda.zeebe.gateway.rest.RestErrorMapper;
+import io.camunda.zeebe.gateway.rest.SearchQueryRequestMapper;
+import io.camunda.zeebe.gateway.rest.SearchQueryResponseMapper;
+import io.camunda.zeebe.gateway.rest.TenantAttributeHolder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@ZeebeRestController
+public class DecisionDefinitionController {
+
+  @Autowired private DecisionDefinitionServices decisionDefinitionServices;
+
+  @PostMapping(
+      path = "/decision-definitions/search",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<DecisionDefinitionSearchQueryResponse> searchDecisionDefinitions(
+      @RequestBody(required = false) final DecisionDefinitionSearchQueryRequest query) {
+    return SearchQueryRequestMapper.toDecisionDefinitionQuery(query)
+        .fold(this::search, RestErrorMapper::mapProblemToResponse);
+  }
+
+  private ResponseEntity<DecisionDefinitionSearchQueryResponse> search(
+      final DecisionDefinitionQuery query) {
+    try {
+      final var tenantIds = TenantAttributeHolder.tenantIds();
+      final var result =
+          decisionDefinitionServices.withAuthentication((a) -> a.tenants(tenantIds)).search(query);
```

Comment:
🔧 You can simplify this with the following:
```java
      final var result = decisionDefinitionServices
          .withAuthentication(RequestMapper.getAuthentication()).search(query);
```

The same can be refactored for the process instance search.

Reply:
right, that's actually doing the same thing

Reply:
it also creates a token, it is meant to be sent to the broker? @tmetzke

Reply:
✅